src/TBuffer.cpp

```diff
@@ -733,7 +733,7 @@ TBuffer::TBuffer(Host* pH)
 , mWrapAt(99999999)
 , mWrapIndent(0)
 , mCursorY(0)
-, mMXP(true)
+, mMXP(false)
```

Comment:
Hang on! Haven't we got the initial states for this and `Host::mServerMXPenabled` swapped around?
- `(bool) Host::mServerMXPenabled` holds the negotiated (bi-directional as it happens) state when set means that the game server has requested (and we have agreed to use MXP) - and **while it is in a `false` state** we must NOT respond to MXP control `CSI`\<mode number>`z` sequences OR convert bytes corresponding to `<`/`>`/`&lt:`/`&gt;` into anything else until that flag gets set.
- `(bool) TBuffer::mMXP` holds the instantaneous MXP processing state (as set by the MXP control sequence but which starts off in the `0` or **On** mode - and in the *simplified* (MXP ~0.3 version ?) that we do - this corresponds to it being `true` and it says whether to convert a subset of `<`/`>`/`&lt:`/`&gt;` as needed by MXP {I cannot recall which of the first or last pair needs to be changed} into the other pair.

In summary I believe it needs to be:
- `mServerMXPenabled` == `false`: do ***nothing*** with control codes *or* text content - ignore the state of `mMXP` - or rather treat it as `false`.

- `mServerMXPenabled` == `true`: respond to control codes that change mMXP state:
    - `mMXP` === `false` do nothing to text content.
    - `mMXP` === `true` change text.

Does this make sense?

Reply:
(I noticed that I used many 'so' to just to highlight some points... I need to be unique in choosing some fancy words...)

Reply:
You say ChatMUD doesn't use MXP - but does it use `<`/`>` ? Ah, it does. Humm let me run a test on an old-skool server that I know doesn't do MXP.

Reply:
well in one of my previous PR change it causes to treat `</>` as mxp literally because I set `mMXP` to true and didn't add an additional safeguard in an `if (mxp)` statement. so now it's  `if(mxp && mphost->mServerMXPenabled)` .. although in this PR change `mServerMXPenabled` might be irrelevant but I want to leave it in there so it will have a significant role to play once IRE updated their rapture engine. although I don't know when.

Reply:
Humm, that works.

Do we know whether IRE will get their MXP handling updated to include negotiation (maybe it is a work around for Mudlet anyhow) ?  If so it would make our code more understandable if the operation was as I outlined I suppose. (At least until we get rid of all the `(bool) Host::aModeIsEnabled` flags associated with option negotiation and make use of the *proper* flags in `cTelnet::hisOptionState` & `cTelnet::myOptionState` ... :grinning: )

I'll dismiss my objections but leave it for @vadi2 to approve it if you do not mind.